plugins/guests/redhat/cap/rhn_register.rb

```diff
@@ -0,0 +1,121 @@
+module VagrantPlugins
+  module GuestRedHat
+    module Cap
+      class RhnRegister
+        # Test that the machine is already registered
+        def self.rhn_register_registered?(machine)
+          true if machine.communicate.execute('/usr/sbin/rhn_check', sudo: true)
+        rescue
+          false
+        end
+
+        # Test that we have rhn installed
+        def self.rhn_register(machine)
+          machine.communicate.test('/usr/sbin/rhn_check --version', sudo: true) &&
+            machine.communicate.test('/usr/sbin/rhnreg_ks --version', sudo: true)
+        end
+
+        # Register the machine using 'rhnreg_ks' command, config is (Open)Struct
+        def self.rhn_register_register(machine, ui)
+          rhn_register_upload_certificate(machine, ui) if machine.config.registration.ca_cert
+          rhn_register_server_url(machine, ui) if machine.config.registration.serverurl
+          command = "rhnreg_ks #{configuration_to_options(machine.config.registration)}"
+          machine.communicate.execute("cmd=$(#{command}); if [ \"$?\" != \"0\" ]; then echo $cmd | grep 'This system is already registered' || (echo $cmd 1>&2 && exit 1) ; fi", sudo: true)
+        end
+
+        # Unregister the machine using 'rhn_unregister.py' resource script
+        def self.rhn_register_unregister(machine)
+          tmp = '/tmp/rhn_unregister'
+          username = machine.config.registration.username
+          password = machine.config.registration.password
+
+          machine.communicate.tap do |comm|
+            # No api call can be made without username/password
+            if username && password
+              serverurl = machine.config.registration.serverurl
+              # Generate the api url
+              serverurl = serverurl.sub(/XMLRPC$/, 'rpc/api')
+              comm.sudo("rm -f #{tmp}", error_check: false)
+              comm.upload(resource('rhn_unregister.py'), tmp)
+              comm.sudo("python #{tmp} -u #{username} -p #{password} -s #{serverurl}")
```

Comment:
For RHN there is no command to delete the system from Spacewalk/Satellite/RHN Classic and the only possibility to do it is with a API call. The script is implementing this feature and will be execute inside the vagrant guest machine.

There is a high chance that `ruby` environment and required libraries are not installed inside the guest machine (not part of RHEL 7.1 box), but `python` environment and required libraries are already installed (required by yum and rhn_register/rhnreg_ks command).